src/analytics.js

```diff
@@ -16,6 +16,7 @@ const {
 	_metrics,
 	Config,
 	Data,
+	Noop,
 } = Altis.Analytics;
 
```

Comment:
I was thinking here you could check the `Noop` value early, then set the empty functions and `return` preventing further execution of the file.

```suggestion
	Noop,
} = Altis.Analytics;

// Expose no-op functions to browser for specific contexts. 
if (Noop) {
	window.Altis.Analytics.updateEndpoint = () => {};
	window.Altis.Analytics.record = () => {};
	return;
}
```

Reply:
Whoops - I made a mistake here, in JS you can't return outside of a function, but simple solution is to wrap the entire thing after the `import`s in a closure function eg:

```js
import ... etc...

( function () {
   // the whole file contents after import goes here
} )();
```